app/models/miq_ae_instance.rb

```diff
@@ -139,6 +148,12 @@ def self.display_name(number = 1)
     n_('Automate Instance', 'Automate Instances', number)
   end
 
+  def self.find_best_match_by(user, relative_path)
+    domain_ids = user.current_tenant.enabled_domains.collect(&:id)
+    joins(:domain).where(:miq_ae_namespaces => {:id => domain_ids}).order("miq_ae_namespaces.priority DESC")
+                  .find_by(arel_table[:relative_path].lower.matches(relative_path.downcase))
```

Comment:
Super-minor, but presentation wise, it would be more readable if the .order were on its own line